resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/backend/minikube/Minikube.scala

```diff
@@ -38,12 +38,50 @@ private[spark] object Minikube extends Logging {
 
   def getMinikubeStatus: MinikubeStatus.Value = {
     val statusString = executeMinikube("status")
+    logInfo(s"Minikube status command output:\n$statusString")
+    // up to minikube version 0.30 use this to check for minikube status
+    val oldMinikube = statusString
       .filter(line => line.contains("minikubeVM: ") || line.contains("minikube:"))
-      .head
-      .replaceFirst("minikubeVM: ", "")
-      .replaceFirst("minikube: ", "")
-    MinikubeStatus.unapply(statusString)
+
+    if (oldMinikube.isEmpty) {
+      if (statusString.length != 4) {
+        return MinikubeStatus.NONE
+      }
+
+      val hostString = statusString.find(_.contains("host: "))
+      val kubeletString = statusString.find(_.contains("kubelet: "))
+      val apiserverString = statusString.find(_.contains("apiserver: "))
+      val kubectlString = statusString.find(_.contains("kubectl: "))
+
+      if (hostString.isEmpty || kubeletString.isEmpty
+        || apiserverString.isEmpty || kubectlString.isEmpty) {
+        MinikubeStatus.NONE
+      } else {
+        val status1 = hostString.get.replaceFirst("host: ", "")
+        val status2 = kubeletString.get.replaceFirst("kubelet: ", "")
+        val status3 = apiserverString.get.replaceFirst("apiserver: ", "")
+        val status4 = kubectlString.get.replaceFirst("kubectl: ", "")
+        if (!status4.contains("Correctly Configured:")) {
+          MinikubeStatus.NONE
+        } else {
+          val stats = List(status1, status2, status3)
+            .map(MinikubeStatus.unapply)
+            .map(_.getOrElse(throw new IllegalStateException(s"Unknown status $statusString")))
+          if (stats.exists(_ != MinikubeStatus.RUNNING)) {
+            MinikubeStatus.NONE
+          } else {
+            MinikubeStatus.RUNNING
+          }
+        }
+      }
+    } else {
+      val finalStatusString = oldMinikube
+        .head
+        .replaceFirst("minikubeVM: ", "")
+        .replaceFirst("minikube: ", "")
+      MinikubeStatus.unapply(finalStatusString)
```

Comment:
match/case here?

Reply:
This code pre-existed I just followed the same style but I can refactor.